qui/tray/domains.py

```diff
@@ -537,6 +541,7 @@ def __init__(self, app_name, qapp, dispatcher, stats_dispatcher):
             _('<b>Qubes Domains</b>\nView and manage running domains.'))
 
         self.tray_menu = Gtk.Menu()
+        self.widget_icon.connect('focus-out-event', self.hide_menu)
```

Comment:
```suggestion
        self.tray_menu.connect('focus-out-event', self.hide_menu)
```

this change to be exact, looks like it does not break anything on X at least.

Reply:
Also this would still require some sort of hacky "hide if visible" modification to the show_menu function, otherwise I can get it funnily stuck on X

Reply:
maybe this is a direction to go to? https://stackoverflow.com/a/3687273

Reply:
nvm, looks like another thing that is broken under Wayland